pkg/detectors/privatekey/privatekey.go

```diff
@@ -85,53 +86,57 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		}
 
 		if verify {
+			g, ctx := errgroup.WithContext(ctx)
 			var (
-				wg                 sync.WaitGroup
 				extraData          = newExtraData()
 				verificationErrors = newVerificationErrors()
 			)
 
 			// Look up certificate information.
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
+			g.Go(func() error {
 				data, err := lookupFingerprint(ctx, fingerprint, s.IncludeExpired)
-				if err == nil {
-					if data != nil {
-						extraData.Add("certificate_urls", strings.Join(data.CertificateURLs, ", "))
-					}
-				} else {
+				if err == nil && data != nil {
+					extraData.Add("certificate_urls", strings.Join(data.CertificateURLs, ", "))
+				} else if err != nil {
 					verificationErrors.Add(err)
 				}
-			}()
+				return nil
+			})
 
-			// Test SSH key against github.com
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
+			// Test SSH key against GitHub.
+			g.Go(func() error {
 				user, err := verifyGitHubUser(ctx, parsedKey)
 				if err != nil && !errors.Is(err, errPermissionDenied) {
 					verificationErrors.Add(err)
 				}
 				if user != nil {
 					extraData.Add("github_user", *user)
 				}
-			}()
+				return nil
+			})
 
-			// Test SSH key against gitlab.com
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
+			// Test SSH key against GitLab.
+			g.Go(func() error {
 				user, err := verifyGitLabUser(ctx, parsedKey)
 				if err != nil && !errors.Is(err, errPermissionDenied) {
 					verificationErrors.Add(err)
 				}
 				if user != nil {
 					extraData.Add("gitlab_user", *user)
 				}
-			}()
+				return nil
+			})
+
+			// Wait for all goroutines to finish or context cancellation.
+			if err := g.Wait(); err != nil {
+				// If context was canceled, set the error and return.
+				if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+					s1.SetVerificationError(err, token)
+					results = append(results, s1)
+					return results, nil
```

Comment:
Does this mean if A succeeds but B is cancelled/exceeds deadline, the data from A won't be reported?